src/handlers/console.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+// import { CodeCell } from '@jupyterlab/cells';
+
+import {
+  IConsoleTracker,
+  ConsolePanel,
+  CodeConsole
+} from '@jupyterlab/console';
+import { DebugSession } from '../session';
+import { IClientSession, WidgetTracker } from '@jupyterlab/apputils';
+import { BreakpointsService } from '../breakpointsService';
+import { CellManager } from '../handlers/cell';
+import { CodeCell } from '@jupyterlab/cells';
+
+export class DebuggerConsoleTracker {
+  constructor(options: DebuggerNotebookTracker.IOptions) {
+    this.breakpointService = options.breakpointService;
+    this.consoleTracker = options.consoleTracker;
+
+    this.consoleTracker.currentChanged.connect(
+      (sender: WidgetTracker<ConsolePanel>, consolePanel: ConsolePanel) => {
+        this.newDebuggerSession(consolePanel, sender);
+      }
+    );
+  }
+
+  consoleTracker: IConsoleTracker;
+  debuggerSession: DebugSession;
+  breakpointService: BreakpointsService;
+  cellManager: CellManager;
+  consoleSession: IClientSession | Boolean;
```

Comment:
Having `consoleSession: IClientSession | null;` would help avoiding the several casts below.